src/controllers/did.ts

```diff
@@ -286,32 +307,27 @@ export class DIDController {
 
 		// handle error
 		if (!result.isEmpty()) {
-			return response.status(StatusCodes.BAD_REQUEST).json({ error: result.array().pop()?.msg });
+			return response.status(StatusCodes.BAD_REQUEST).json({
+				error: result.array().pop()?.msg,
+			} satisfies ValidationErrorResponseBody);
 		}
 
 		// handle request params
-		const { did, service, verificationMethod, authentication } = request.body as {
-			did: string;
-			service: Service[];
-			verificationMethod: VerificationMethod[];
-			authentication: string[];
-		};
-		let updatedDocument: DIDDocument;
+		const { did, service, verificationMethod, authentication } = request.body as UpdateDidRequestBody;
+		// Get the didDocument from the request if it's placed there
+		let updatedDocument: DIDDocument | undefined = request.body.didDocument ;
```

Comment:
I'd refactor this to remove mutation of the request body parameter, namely by creating a deep copy + assembling changes isolated to that copy.

That means switching to a top-level constant definition, with readonly properties.

Let's keep that for the next set of improvements.